lib/irida/pipeline.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Irida
+  # Class to store pipeline values
+  class Pipeline
+    attr_accessor :name, :description, :metadata, :type, :type_version,
+                  :engine, :engine_version, :url, :execute_loc,
+                  :version, :schema_loc, :schema_input_loc
+
+    def initialize(entry, version, schema_loc, schema_input_loc)
+      @name = entry['name']
+      @description = entry['description']
+      @metadata = { workflow_name: name, workflow_version: version }
+      @type = nil
```

Comment:
**nitpick (blocking):** Can you set these `nil` fields to the values that were hardcoded previously in the submissions controller as those are the defaults and should be set here. Otherwise submission will fail on WES.